### DISCUSSION TITLE
Historical graph

### DISCUSSION THREAD

**Discussion by SebastienSaco** [frangoteam/FUXA#355] [Q&A] [answered]
Hello,

Is it possible to have an historical chart with more than one week of history (a year for exemple) ?

Is there a way of importing tags from any kind of formatted file (json,xml,xsl...), instead of doing that from the UI ?

Great work and thanks for your help.

**Comment by unocelli** [comment, marked as answer]
Hi,
you can select what you want with the daterange 
![image](https://user-images.githubusercontent.com/48684550/168175021-3ef8a926-d691-41ac-870a-8508a9a4588f.png)

you can export the devices with the tags in a json file, edit it and then import it.
if you are familiar with json it should not be a problem to understand how it is structured, it is important that it respects the ids which must be unique and which are then referenced with the controls of the gui.

![image](https://user-images.githubusercontent.com/48684550/168175378-30671702-e3f3-4626-b554-e687b62cf9d2.png)

**Comment by florentgelas** [comment]
Hello,
How are save the historical data please.
Is it possible to export historical data from a chart or something else?

Thanks

Florent

**Reply by unocelli** [reply]
Hi,
In the last version of master branch you can configure a table and export the data in CSV format

**Reply by joniyes** [reply]
Hi, could you show the menu location to save data in CSV ?

**Reply by unocelli** [reply]
Configure:
Type: History table
Layout: Date Range = enabled

**Reply by joniyes** [reply]
Thank you, now i can save data to csv
